ports/geckolib/properties.mako.rs

```diff
@@ -853,76 +853,84 @@ fn static_assert() {
 
         unsafe {
             // Prevent leaking of the last element we did set
-            Gecko_SetNullImageValue(&mut self.gecko.mImage.mLayers.mFirstElement.mImage);
+            for image in &mut self.gecko.mImage.mLayers {
+                Gecko_SetNullImageValue(&mut image.mImage)
+            }
         }
 
-        self.gecko.mImage.mImageCount = cmp::max(1, self.gecko.mImage.mImageCount);
-        if let Some(image) = image.0 {
-            match image {
-                Image::LinearGradient(ref gradient) => {
-                    let stop_count = gradient.stops.len();
-                    if stop_count >= ::std::u32::MAX as usize {
-                        warn!("stylo: Prevented overflow due to too many gradient stops");
-                        return;
-                    }
-
-                    let gecko_gradient = unsafe {
-                        Gecko_CreateGradient(NS_STYLE_GRADIENT_SHAPE_LINEAR as u8,
-                                             NS_STYLE_GRADIENT_SIZE_FARTHEST_CORNER as u8,
-                                             /* repeating = */ false,
-                                             /* legacy_syntax = */ false,
-                                             stop_count as u32)
-                    };
-
-                    // TODO: figure out what gecko does in the `corner` case.
-                    if let AngleOrCorner::Angle(angle) = gradient.angle_or_corner {
-                        unsafe {
-                            (*gecko_gradient).mAngle.set(angle);
+        self.gecko.mImage.mImageCount = cmp::max(self.gecko.mImage.mLayers.len() as u32,
+                                                 self.gecko.mImage.mImageCount);
+
+        // TODO: pre-grow the nsTArray to the right capacity
+        // otherwise the below code won't work
+        for (image, geckoimage) in images.0.into_iter().zip(self.gecko.mImage.mLayers.iter_mut()) {
```

Comment:
Github doesn't show this diff well, but the below code is simply indented one level with a change to the `Gecko_SetGradientImageValue` line